packages/block-editor/src/store/selectors.js

```diff
@@ -1986,7 +1986,7 @@ export const getInserterItems = createSelector(
 				isDisabled: false,
 				utility: 1, // Deprecated.
 				frecency,
-				content: reusableBlock.content.raw,
+				content: reusableBlock.content?.raw,
```

Comment:
Per the symbolicated stack trace:

```
gutenberg/packages/block-editor/src/store/selectors.js:1988:buildReusableBlockInserterItem
gutenberg/packages/block-editor/src/store/selectors.js:1998:createSelector$argument_0
gutenberg/node_modules/rememo/rememo.cjs:270:callSelector
gutenberg/packages/data/src/redux-store/index.js:237:boundSelector
gutenberg/packages/block-editor/src/components/inserter/menu.native.js:70:useSelect$argument_0
```

Seems this is the line that produces the exception.

Reply:
@fluiddot, is that stack trace from a recent crash? I've been looking at a few symbolicated stack traces from recent crashes (from after `23.0`, which included https://github.com/WordPress/gutenberg/pull/53721) but they've all been a bit more vague than what you're seeing. For example: 

```
This error is located at:
    in C
    in y
    in R
    in WithPreferredColorScheme(R)
    in Unkn..., stack:
/workdir/gutenberg/packages/block-editor/src/store/actions.js:933:<anonymous>
/workdir/gutenberg/packages/block-editor/src/store/actions.js:860:registry.batch$argument_0
/workdir/gutenberg/node_modules/rememo/rememo.cjs:157:getCache
/workdir/gutenberg/packages/data/src/redux-store/index.js:233:bindSelector
null:null:null
/workdir/gutenberg/packages/data/src/components/use-select/index.js:151:updateValue
/workdir/gutenberg/packages/data/src/registry.js:293:registerStoreInstance$argument_1
/workdir/gutenberg/packages/data/src/registry.js:125:__unstableMarkListeningStores
/workdir/gutenberg/packages/data/src/components/use-select/index.js:39:Store
/workdir/gutenberg/packages/data/src/components/use-select/index.js:17:<global>
/workdir/gutenberg/packages/data/src/components/use-select/index.js:106:updateStores
/workdir/gutenberg/packages/data/src/components/use-select/index.js:196:useMappingSelect
/workdir/gutenberg/packages/block-editor/src/components/inner-blocks/use-nested-settings-update.js:64:useSelect$argument_0
```

Reply:
Yes, I used the stack trace from [this Sentry event](https://a8c.sentry.io/share/issue/25c040f835554d5fa64a9e574687d585/) and picking a recent event for version `23.2.2 (GBM: 1.103.3)`:

```
anonymous@1:3088153
anonymous@1:3087780
o@1:1722534
i@1:1617316
anonymous@1:3129196
anonymous@1:1665759
__unstableMarkListeningStores@1:1612035
anonymous@1:1612238
w@1:1665550
anonymous@1:1665458
b@1:1666351
anonymous@1:1666493
C@1:3128556
Or@1:381186
Ma@1:421763
zi@1:408435
Ri@1:408337
Pi@1:408221
ki@1:405721
xt@1:370846
Ce@1:422136
Ne@1:363989
Me@1:364262
receiveTouches@1:416972
value@1:165707
anonymous@1:164207
value@1:165153
value@1:164165
```
⬇️ 
```
gutenberg/packages/block-editor/src/store/selectors.js:1988:buildReusableBlockInserterItem
gutenberg/packages/block-editor/src/store/selectors.js:1998:createSelector$argument_0
gutenberg/node_modules/rememo/rememo.cjs:270:callSelector
gutenberg/packages/data/src/redux-store/index.js:237:boundSelector
gutenberg/packages/block-editor/src/components/inserter/menu.native.js:70:useSelect$argument_0
gutenberg/packages/data/src/components/use-select/index.js:134:registry.__unstableMarkListeningStores$argument_0
gutenberg/packages/data/src/registry.js:123:__unstableMarkListeningStores
gutenberg/packages/data/src/registry.js:204:<anonymous>
gutenberg/packages/data/src/components/use-select/index.js:133:updateValue
gutenberg/packages/data/src/components/use-select/index.js:180:<anonymous>
gutenberg/packages/data/src/components/use-select/index.js:198:useMappingSelect
gutenberg/packages/data/src/components/use-select/index.js:287:useSelect
gutenberg/packages/block-editor/src/components/inserter/menu.native.js:53:InserterMenu
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:3525:renderWithHooks
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:7743:beginWork$1
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:7304:performUnitOfWork
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:7297:workLoopSync
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:7279:renderRootSync
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:6975:performSyncWorkOnRoot
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:2145:flushSyncCallbacks
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:8462:batchedUpdatesImpl
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:1106:batchedUpdates
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:1137:_receiveRootNodeIDEvent
gutenberg/node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-prod.js:1209:ReactNativePrivateInterface.RCTEventEmitter.register$argument_0.receiveTouches
gutenberg/node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:427:__callFunction
gutenberg/node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:113:__guard$argument_0
gutenberg/node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:368:__guard
gutenberg/node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:112:callFunctionReturnFlushedQueue
```

Reply:
That's interesting, thanks for confirming! I wonder why there's such a variation.

Reply:
> I wonder why there's such a variation.

It might be related to the Sentry event, I noticed that we have different ones pointing to similar errors:
- [`TypeError: Cannot read property 'raw' of undefined ` (only affecting newer version, 23.0 and above)](https://a8c.sentry.io/share/issue/25c040f835554d5fa64a9e574687d585/)
- [`TypeError: undefined is not an object (evaluating 'n.title.raw')` (only affecting past versions 22.8 and below)](https://a8c.sentry.io/share/issue/fd7e35fde101477caae64ba2ee5a40da/)

@SiobhyB I'm wondering which Sentry event you used to symbolicate the stack trace.

Reply:
@fluiddot, I've been using multiple different events from the _newer_ crash, as the older one went away following https://github.com/WordPress/gutenberg/pull/53721.

Reply:
That's interesting, in that case, and as you pointed out, seems the Sentry event might be grouping exceptions caused by different locations 😬. I presume this fix will solve most of the exceptions we are getting, so let's see after releasing this fix if we still get the event in newer versions 🤞 .